src/metabase/driver/druid/query_processor.clj

```diff
@@ -730,14 +730,26 @@
 
 (def ^:private druid-ts-format (tformat/formatters :date-time))
 
+(defn- parse-timestamp
```

Comment:
I don't think this is what we want here, for a few reasons. Elsewhere in the app, we use JodaTime's parsers. Mostly this is due to them being threadsafe and not suffering from the performance issues and other gotchas that SimpleDateFormat suffers from. My guess is what you've got here is pretty slow.

Another reason to use the JodaTime stuff is that it lets us control the timezone. Using the built-in java stuff will discard the original timezone and convert it to the JVM's timezone. Not sure if that's a big deal here or not since we're consuming this internally, but parsing it in JodaTime would allow us to keep it in UTC, rather than converting it to something random.

Reply:
I parse to Date directly, so that it's the same format as what comes from ODBC. Based on (very provisional) benchmarks it's marginally faster than going str->Joda->Date and more importantly produces less garbage, but it's not a super big win, so I'm totally fine with going back to Joda.

Reply:
Did you compare it to using `tformat/parse`? I would expect that parse function to be quite a bit slower. It uses formatters in an arbitrary order, choosing the first one that succeeds. If you're lucky and the format you use is at the beginning it might be ok, but it's a hashmap, so the order is not stable across machines. I wrote [this](https://github.com/metabase/metabase/blob/master/src/metabase/util.clj#L883) to put the more likely candidates first, but if you already know the format of the timestamp, it's faster to just use that format directly.